### PR TITLE
poc(Button): display svg icon path on button hover

### PR DESCRIPTION
**Pull request by miroslavstastny**
This is just an experiment to learn how `Button` with `Icon` is supposed to be used in order to implement button which contains an icon which is 'filled' on hover.

### Requirements (@200%):

  | normal | :hover
-- | -- | --
light |  image  |  image
dark | image | image
hc | image  | image

### Proposed implementation concerns

- Proposed implementation works fine for light and dark theme, I do not know how to implement that for HC theme.
- It uses static class name (`ui-icon__filled`)

### CODE REVIEW

File: src/themes/teams/components/Icon/svg/icons/bold.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	           c0.5,0.2,0.9,0.5,1.2,0.9c0.7,0.7,1,1.6,1,2.6c0,1.4-0.5,2.5-1.6,3.5C19.4,23.1,18,23.5,16.4,23.5z M14.5,20.2h1.2
+	           c0.6,0,1.1-0.1,1.4-0.4c0.3-0.2,0.4-0.7,0.4-1c0-0.4-0.1-0.7-0.4-0.9c-0.3-0.3-0.9-0.4-1.4-0.4h-1.2V20.2z M14.5,14.1h0.9
+	           c0.5,0,0.9-0.1,1.1-0.4c0.3-0.2,0.5-0.6,0.5-0.9c0-0.3,0-1-1.8-1h-0.7V14.1z"
+        />
+      </g>
+    </svg>
+  ),
+  styles: {
+    svg: () => ({
+      color: 'inherit',
```
Comment (miroslavstastny) [inline src/themes/teams/components/Icon/svg/icons/bold.tsx:36]: This works better with button hover, see circular button example with this icon vs call icon

File: src/themes/teams/components/Icon/svg/icons/bold.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react'
+import * as cx from 'classnames'
+import { TeamsSvgIconSpec } from '../types'
+
+export default {
+  icon: ({ classes }) => (
+    <svg viewBox="0 0 32 32" role="presentation" className={classes.svg}>
+      <g className="icons-default-fill">
```
Comment (miroslavstastny) [inline src/themes/teams/components/Icon/svg/icons/bold.tsx:8]: Not sure if we need this class for anything.

File: src/themes/teams/components/Icon/iconVariables.ts [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
-  [key: string]: string | number | undefined
+  [key: string]: string | number | boolean | undefined
 
   color?: string
   backgroundColor?: string
   borderColor?: string
   horizontalSpace: string
   margin: string
   secondaryColor: string
+  filled?: boolean
```
Comment (kuzhelov) [inline src/themes/teams/components/Icon/iconVariables.ts:13]: probably, we should think about making this part of the Icon's interface

File: docs/src/examples/components/Button/Usage/index.tsx [hunk 4]
```diff
@@ -9,6 +9,11 @@ const Usage = () => (
       description="A button used in cards is a &quot;tinted&quot; version of a default button."
       examplePath="components/Button/Usage/ButtonUsageExample"
     />
+    <ComponentExample
+      title="Toolbar button"
+      description="Icon paths change on hover"
```
Comment (kuzhelov) [inline docs/src/examples/components/Button/Usage/index.tsx:14]: nit: just for the sake of consistency, there should be dot at the end ;)

**Review by kuzhelov — APPROVED** [review]
do like the approach - the only problem, probably, an apparent one, is that there are quite a lot of indirections involved here. At the same time, worth to notice that this PR reduces, arguably, their amount to the best possible one. Also, it seems that there are some ways to further encapsulate idea of this approach under more convenient generic API exposed by utility methods

**Comment by miroslavstastny** [comment]
@codepretty - could you please review this to verify it is in line with your changes/plans on SVG icons?

File: docs/src/examples/components/Button/Usage/ButtonExampleToolbarButton.shorthand.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+const ButtonExampleToolbarButton = () => (
+  <Button
+    icon={{
+      name: 'bold',
+      size: 'big',
+    }}
+    iconOnly
+    text
+    styles={{
+      '&:hover .ui-icon__filled': {
```
Comment (codepretty) [inline docs/src/examples/components/Button/Usage/ButtonExampleToolbarButton.shorthand.tsx:13]: Wouldn't you also need to hide the unfilled version?

File: src/themes/teams/components/Icon/svg/icons/bold.tsx (hunk 2 above)
Comment (codepretty) [reply src/themes/teams/components/Icon/svg/icons/bold.tsx:8]: We don't need this for icons in Stardust. I am going to be importing all the icons without this class

File: src/themes/teams/components/Icon/iconStyles.ts [hunk 6]
```diff
@@ -122,6 +122,9 @@ const iconStyles: ComponentSlotStylesInput<IconProps, any> = {
   path: getSvgStyle('path'),
 
   secondaryPath: getSvgStyle('secondaryPath'),
+
+  unfilled: getSvgStyle('unfilled'),
```
Comment (codepretty) [inline src/themes/teams/components/Icon/iconStyles.ts:126]: I am implementing this in my PR where we won't need to have these styles defined in each SVG. I also named them filled & outline. Do you think filled & unfilled works better?

Comment (miroslavstastny) [reply src/themes/teams/components/Icon/iconStyles.ts:126]: I think `filled` & `outline` are fine, `unfilled` was used in legacy client.

**Comment by miroslavstastny** [comment]
Closing this for now, will create a new PR once #478 is merged.

**Pull request closed without merging**